documentation/modules/managing/con-producer-config-properties.adoc

```diff
@@ -0,0 +1,203 @@
+The producer uses the address to discover and connect to all brokers in the cluster.
+Use a comma-separated list to specify two or three addresses in case a server is down, but it’s not necessary to provide a list of all the brokers in the cluster.
+<2> (Required) Serializer to transform the message keys to bytes prior to them being sent to a broker.
+<3> (Required) Serializer to transform message values to bytes prior to them being sent to a broker.
+<4> (Optional) The logical name for the client, which is used in logs and metrics to identify the source of a request.
+<5> (Optional) The codec for compressing messages, which are sent and stored in compressed format and then decompressed when reaching a consumer.
+Compression is useful for improving throughput and reducing the load on storage, but might not be suitable for low latency applications where the cost of compression or decompression is prohibitive.
+
+== Data durability
+
+You can apply greater data durability, to minimize the likelihood that messages are lost, using message delivery acknowledgments.
+
+----
+# ...
+acks=all <1>
+# ...
+----
+
+<1> Specifying `acks=all` forces a partition leader to replicate messages to all followers before
+acknowledging that the message request was successfully received.
+Because of the additional checks, `acks=all` increases the latency between the producer sending a message and receiving acknowledgment.
+
+In addition to using `acks=all`, you must configure the topics receiving the messages to have more than one in-sync replica.
+A typical starting point is to have a topic replication factor of 3, with two in-sync replicas on other brokers.
+In this configuration, the producer can continue unaffected if a single broker is unavailable.
+If a second broker becomes unavailable, the producer won’t receive acknowledgments and won’t be able to produce more messages.
+
+.Topic configuration to support `acks=all`
+----
+# ...
+min.insync.replicas=2 <1>
+# ...
+----
+<1> Use `2` in-sync replicas. The default is `1`.
+
+== Ordered delivery
+
+Idempotent producers avoid duplicates as messages are delivered exactly once.
+IDs and sequence numbers are assigned to messages to ensure the order of delivery, even in the event of failure.
+If you are using `acks=all` for data consistency, enabling idempotency makes sense for ordered delivery.
+
+.Ordered delivery with idempotency
+----
+# ...
+enable.idempotence=true <1>
+max.in.flight.requests.per.connection=5 <2>
+acks=all <3>
+retries=2147483647 <4>
+# ...
+----
+<1> Set to `true`.
+<2> Specify the number of allowed in-flight requests to a maximum of 5.
+<3> Set to `all`.
+<4> Set the number of attempts to resend a failed message request.
+
+If the performance cost is too great, and you are not using `acks=all`,
+set the number of in-flight (unacknowledged) requests to 1 to preserve ordering.
```

Comment:
I don't think makes sense ... setting in-flight messages to 1 will slow down the producer. So it does not fit for me with the _If the performance cost is too great_. Maybe something like _If performance is not the main priority_ or something like that?

Reply:
Yes, it's related to the next example.
I've changed to:

If you are not using `acks=all` and idempotency because of the performance cost, set the number of in-flight (unacknowledged) requests to 1 to preserve ordering.